execution_workflows/IsoSCM/modules/star_alignment.nf

```diff
@@ -0,0 +1,44 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+def options    = initOptions(params.options)
+
+
+/*
+    Perform STAR alignment containing three steps
+    1. Get the fastq file of a sample bam file
+    2. Run STAR alignment of the sample bam file
+    3. Get the bai file of the new aligned bam file
+*/
+process STAR_ALIGNMENT {
+    publishDir "${params.outdir}/isoscm/aligned_bam_files/${sample}.${strand}", mode: params.publish_dir_mode
+    container "docker.io/apaeval/isoscm:latest"
+    label 'process_high'
+
+    input:
+    tuple val(sample), path(bam), val(strand), path(star_index_file), val(star_genome_index_indicator)
+    
+    output:
+    tuple val(sample), val(strand), path(star_out_bam), path(star_out_bai), emit: ch_aligned_bam_files
+
+    script:
+    fastq_file = sample + ".fastq"
+    star_out_bam = sample + ".Aligned.sortedByCoord.out.bam"
+    star_out_bai = sample + ".Aligned.sortedByCoord.out.bam.bai"
+    """
+    samtools bam2fq $bam > $fastq_file
+    
```

Comment:
1- I think this should be a separate process. If have option to set as temporary (i.e. delete after alignment complete) that would be ideal to save disk space. I think should also suffix with .fastq.gz so files are automatically gzipped (this happens with `samtools fastq`, not sure about `samtools bam2fq` command isn't in latest documentation)

2- Have to be very careful here I think with how we pull from the BAM too. As it stands, all reads will be pulled to a single FASTQ file. This is fine for single-end alignments, but for paired-end alignments you will lose the paired information and align all reads individually (STAR asks you to pass paired-end alignments consecutively on the command line). I suspect if we want to extract paired-end reads properly we will also have to sort the input BAM file by name first (e.g. as for [`samtools fastq`](http://www.htslib.org/doc/samtools-fasta.html)). Would need to add option to config asking user to specify whether input alignments are single-end / paired-end too

![image](https://user-images.githubusercontent.com/49978382/159703556-c915b0e8-e600-4418-82b9-1badaeebf1c9.png)

Reply:
Thanks for pointing this out @SamBryce-Smith, I wasn't familiar with single and paired end reads and this was so helpful! :D I've updated the workflow to include single/paired information in the sample sheet so that each sample runs two different fastq and alignment steps based on the information